### PR TITLE
Move jlink parameters into project.properties 

### PR DESCRIPTION
**Pull request by tresf**
`JLink.java` is the helper class for bundling Java into the Desktop installers and is capable of multiple vendors, platform and architectures including cross-packaging.

Unfortunately, JLink currently require source-code-changes for vendor and platform changes.
https://github.com/qzind/tray/blob/361a9beb8ad55510ee934ac92dcd80c534562bfc/src/qz/build/JLink.java#L29-L36

This PR will move these properties into `project.properties` so that they can be bumped by the build system (ant), e.g:

```properties
jlink.java.vendor="BellSoft"
# Java vendor to bundle into software (e.g. "11.0.17+7")
jlink.java.version="11.0.17+7"
# Java garbage collector flavor to use (e.g. "hotspot|openj9")
jlink.java.gc="hotspot"
# Java garbage collector version to use (openj9 only, e.g. "0.35.0")
jlink.java.gc.version="0.35.0"
```

### CODE REVIEW

File: src/qz/build/JLink.java [hunk 1]
```diff
@@ -103,53 +110,20 @@ private static boolean needsDownload(Version want, Version installed) {
...
-            case BELL:
-                fileExt = platform.equals("linux") ? "tar.gz" : "zip";
-                break;
-            case ADOPT:
-            default:
-                fileExt = platform.equals("windows") ? "zip" : "tar.gz";
-        }
-
-        String url = VendorUrlPattern.format(javaVendor, arch, platform, gcEngine, JAVA_MAJOR, JAVA_VERSION, JAVA_VERSION_FILE, fileExt);
+        String javaVersionUnderscore = javaSemver.toString().replaceAll("\\+", "_");
```
Comment (tresf) [inline src/qz/build/JLink.java:116]: I believe `javaVersionUnderscore` can safely be removed here.

File: src/qz/build/JLink.java [hunk 2]
```diff
@@ -22,44 +26,56 @@
...
     private static final Logger log = LogManager.getLogger(JLink.class);
-    private static final String JAVA_VENDOR = "BellSoft";
-    private static final String JAVA_VERSION = "11.0.17+7";
-    private static final String JAVA_MAJOR = JAVA_VERSION.split("\\.")[0];
-    private static final String JAVA_MINOR = JAVA_VERSION.split("\\.")[1];
-    private static final String JAVA_PATCH = JAVA_VERSION.split("\\.|\\+|-")[2];
-    private static final String JAVA_VERSION_FILE = JAVA_VERSION.replaceAll("\\+", "_");
-    private static final String JAVA_DEFAULT_GC_ENGINE = "hotspot";
-    private static final String JAVA_DEFAULT_ARCH = VendorArch.ADOPT_AMD64.use;
+    public static final Vendor JAVA_VENDOR = Vendor.BELLSOFT;
```
Comment (tresf) [inline src/qz/build/JLink.java:32]: These should be renamed to `DEFAULT_...` to match the others.

**Comment by tresf** [comment]
All obvious issues address, this should be ready for a second set of eyes.

In regards to testing, basic functionality should be sufficient, since this only affects the technique used to download the bundled JVM.

File: src/qz/build/jlink/Parsable.java [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+                        if (match.equalsIgnoreCase(value)) {
+                            return parsable;
+                        }
+                    }
+                } catch(NoSuchFieldException | IllegalAccessException | ClassCastException e) {
+                    log.warn("Parsable enums must have a 'public String[] matches' field", e);
+                }
+            }
+        }
+        log.warn("Could not parse {} as a valid {}} value", value, enumType.getSimpleName());
```
Comment (bberenz) [inline src/qz/build/jlink/Parsable.java:47]: There an extra `}` on the second formatting that should probably be removed

**Review by bberenz — APPROVED** [review]

File: src/qz/build/jlink/Parsable.java (hunk 3 above)
Comment (tresf) [reply src/qz/build/jlink/Parsable.java]: Good catch. :)

**Pull request merged**